src/generator/AutoRest.Ruby.Azure/TemplateModels/AzureMethodTemplateModel.cs

```diff
@@ -119,7 +119,7 @@ public override string ResponseGeneration()
                 if (ReturnType.Body is CompositeType)
                 {
                     CompositeType compositeType = (CompositeType)ReturnType.Body;
-                    if (compositeType.Extensions.ContainsKey(AzureExtensions.PageableExtension))
+                    if (compositeType.Extensions.ContainsKey(AzureExtensions.PageableExtension) && this.Extensions.ContainsKey("nextMethodName"))
```

Comment:
>&& this.Extensions.ContainsKey("nextMethodName") [](start = 96, length = 48)

From line 124, looks like we don't need `&& this.Extensions.ContainsKey("nextMethodName")` condition or we don't need line 124